app/models/test.rb

```diff
@@ -6,10 +6,12 @@ class Test < ApplicationRecord
   has_many :tests_users, dependent: :destroy
   has_many :users, through: :tests_users
   
-  def self.all_tests_by_category(category)
-    Test.joins(:category).where(categories: { name: category })
-                                              .order(id: :desc)
-                                              .pluck(:title)
-  end
-    
+  scope :easy, -> {where(level: 0..1)}
+  scope :intermediate, -> {where(level: 2..4)}
+  scope :hard, -> {where(level:5..Float::INFINITY)}
+  scope :all_tests_by_category, -> (category) {Test.joins(:category).where(categories: {name:category}).order(id: :desc)}
+  
+  validates :level, numericality: {only_integer: true}
+  validates :title, uniqueness: { scope: :level, message: 'uniqueness error' }
```

Comment:
[в документации](https://guides.rubyonrails.org/active_record_validations.html#uniqueness) к хелперу `uniqueness` сказано следующее:
> It does not create a uniqueness constraint in the database, so it may happen that two different database connections create two records with the same value for a column that you intend to be unique. To avoid that, you must create a unique index on that column in your database.

другими словами, чтобы эта была настоящая уникальность (не только на уровне приложения, но и на уровне базы), нужно в базе создать уникальный индекс

про правильную миграцию можно посмотреть к примеру на [СО](https://stackoverflow.com/questions/32927262/create-migration-with-uniqueness-to-a-scope)